services/brig/src/Brig/User/API/Search.hs

```diff
@@ -92,6 +93,7 @@ search ::
   Maybe (Range 1 500 Int32) ->
   (Handler r) (Public.SearchResult Public.Contact)
 search searcherId searchTerm maybeDomain maybeMaxResults = do
+  ensurePermissionsOrPersonalUser searcherId [SearchContacts]
```

Comment:
This could be moved into `searchLocally`, which already looks up the team id. Federated searching has its own policies, maybe this check isn't relevant there.

Reply:
i agree that's a possible optimization, but i think we should also disallow remote searches for ephemeral users.  i'll leave a comment in the code, ok?